helix-core/src/register.rs

```diff
@@ -1,20 +1,63 @@
-use crate::Tendril;
-use once_cell::sync::Lazy;
-use std::{collections::HashMap, sync::RwLock};
-
-// TODO: could be an instance on Editor
-static REGISTRY: Lazy<RwLock<HashMap<char, Vec<String>>>> =
-    Lazy::new(|| RwLock::new(HashMap::new()));
-
-/// Read register values.
-pub fn get(register_name: char) -> Option<Vec<String>> {
-    let registry = REGISTRY.read().unwrap();
-    registry.get(&register_name).cloned() // TODO: no cloning
+use std::collections::HashMap;
+
+#[derive(Debug)]
+pub struct Register {
+    name: char,
+    values: Vec<String>,
```

Comment:
Can we change this to use `Nested` or something similar?

Reply:
Just a suggestion, not a necessary change.

Reply:
I'll consider that! Thanks